docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,464 @@
+  If `X` is the set of all actions in `G`, then `X` is omitted.
+
+Definition »Reduction of an interoperable consistent causality graph«
+  For an `X`\ -interoperable consistent causality graph `G`, there exists a unique minimal `X`\ -interoperable consistent causality graph `reduce`:sub:`X`\ `(G)` with the same vertices and the edges being a subset of `G`.
+  `reduce`:sub:`X`\ `(G)` is called the `X`\ -**reduction** of `G`.
+  As before, `X` is omitted if it contains all actions in `G`.
+
+Since interoperable causality graphs are acyclic, their vertices can be sorted topologically and the resulting list is again a causality graph, where every vertex has an outgoing edge to all later vertices.
+If the original causality graph is `X`\ -consistent, then so is the topological sort, as topological sorting merely adds edges.
+
+
+From interoperable causality graphs to ledgers
+==============================================
+
+Interoperable causality graphs `G` are linked to ledgers `L` in the DAML Ledger Model via topological sort and reduction.
+
+* Given an interoperable causality graph `G`,
+  drop the incoming and outgoing ledger annotations and all transfer vertices,
+  topologically sort the transaction vertices,
+  and extend the resulting list of transactions with the requesters to obtain a sequence of commits `L`.
+
+* Given a sequence of commits `L`,
+  use the transactions as vertices and add an edge from `tx1` to `tx2` whenever `tx1`\ 's commit precedes `tx2`\ 's commit in the sequence.
+  Then add transfer vertices and incoming and outgoing ledger annotations as needed and connect them with edges to the transaction vertices.
+
+This link preserves consistency only to some extent.
+Namely, if an interoperable causality graph is interoperable consistent for a contract `c`, then the corresponding ledger is consistent for the contract `c`, too.
+However, an interoperable-consistent causality graph does not yield a consistent ledger because key consistency may be violated.
+Conversely, a consistent ledger does not talk about the incoming and outgoing ledger annotations and therefore cannot enforce that the annotations are consistent.
+
+.. _ledger-aware-projection:
+
+Ledger-aware projection
+***********************
+
+A Participant Node maintains a local ledger for each party it hosts and the Transaction Service outputs a topological sort of this local ledger.
+When the Participant Node hosts the party on several ledgers, this local ledger is an interoperable causality graph.
+This section defines the ledger-aware projection of an interoperable causality graph, which yields such a local ledger.
+
+Definition »Y-labelled action«
+  An action with incoming and outgoing ledger annotations is **Y-labelled** for a set `Y`
+  if its incoming or outgoing ledger annotation is an element of `Y`.
+
+Definition »Ledger-aware projection for transactions«
```

Comment:
`A` should be `P`. Changed.

Reply:
Can you clarify what confuses you about maximal elements?

`Act` is a subset of all the subactions of the transaction `tx`. All subactions of `tx` are ordered w.r.t. the subaction relation. Now, `act` is a maximal element of `Act` if there is no other `act'` in `Act` of which `act` is a sub-action. (https://en.wikipedia.org/wiki/Maximal_and_minimal_elements). Now, `the maximal elements` are all those maximal elements of `Act`. If they contain another subaction whose labelling is not in `Y`, then that subaction is nevertheless part of the ledger-aware projection, because an action contains all its subactions.

Do you think I should expand the explanation?

Reply:
Added an explanation as discussed offline.